pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -1724,3 +1757,45 @@ func (service *SecurityPolicyService) getNamespaceUID(ns string) (nsUid types.UI
 	namespace_uid := namespace.UID
 	return namespace_uid
 }
+
+func (service *SecurityPolicyService) buildRulePortString(port *v1alpha1.SecurityPolicyPort) string {
+	protocol := string(port.Protocol)
+	if port.EndPort != 0 {
+		return fmt.Sprintf("%s.%s.%d", protocol, (port.Port).String(), port.EndPort)
+	}
+	return fmt.Sprintf("%s.%s", protocol, (port.Port).String())
+
+}
+
+func (service *SecurityPolicyService) buildRulePortsString(ports *[]v1alpha1.SecurityPolicyPort, direction string) string {
+	if ports == nil || len(*ports) == 0 {
+		return "all"
```

Comment:
For there is no ports specified case,  we should generate all-ingress-allow or all-egress-allow. 
So, we can set portsString := all instead of returning here, is it?

Reply:
Yes you are right! Fixed and added a UT for the case.